StylableWinFormsControls/StylableWinFormsControls/Utilities/CustomFormBorderStyleHandler.cs

```diff
@@ -0,0 +1,117 @@
+﻿using StylableWinFormsControls.Controls;
+
+namespace StylableWinFormsControls.Utilities
+{
+    /// <summary>
+    /// provides drawing logic for borders on a form
+    /// </summary>
+    internal static class CustomFormBorderStyleHandler
+    {
+        /// <summary>
+        /// As the drawing is called often, we do not want to create the Brush every time so we cache them
+        /// </summary>
+        private static readonly Dictionary<Color, Brush> BRUSH_CACHE = new Dictionary<Color, Brush>();
+
+        /// <summary>
+        /// draw a border
+        /// </summary>
+        /// <param name="g">the graphics object to draw on</param>
+        /// <param name="borderColor">the default color for the border. may be changed slightly (e.g. for 3D effects)</param>
+        /// <param name="f">the form to get the settings from</param>
+        public static void DrawBorder(this Graphics g, Color borderColor, Form f)
+        {
+            g.DrawBorder(borderColor, f.FormBorderStyle);
+        }
+
+        /// <summary>
+        /// draw a border
+        /// </summary>
+        /// <param name="g">the graphics object to draw on</param>
+        /// <param name="f">the form to get the settings from</param>
+        public static void DrawBorder(this Graphics g, StylableMdiChildForm f)
+        {
+            g.DrawBorder(f.BorderColor, f);
+        }
+
+        /// <summary>
+        /// draw a border
+        /// </summary>
+        /// <param name="g">the graphics object to draw on</param>
+        /// <param name="borderColor">the default color for the border. may be changed slightly (e.g. for 3D effects)</param>
+        /// <param name="style">the Border Style</param>
+        /// <param name="topOffset">offset at the top. Can be used to allow space for a title bar</param>
+        /// <param name="bottomOffset">offset at the bottom</param>
+        /// <param name="leftOffset">offset at the left</param>
+        /// <param name="rightOffset">offset at the right</param>
+        public static void DrawBorder(this Graphics g, Color borderColor, FormBorderStyle style, int topOffset = 0, int bottomOffset = 0, int leftOffset = 0, int rightOffset = 0)
+        {
+            switch (style)
+            {
+                case FormBorderStyle.Fixed3D:
+                //TODO: Implement
```

Comment:
Is this TODO intended to be included in this PR?

Reply:
Maybe not. I wanted to start providing support for MDI but as I currently have no usecase myself it is not high priority.